components/pkg-export-docker/src/main.rs

```diff
@@ -98,11 +111,51 @@ fn start(ui: &mut UI) -> Result<()> {
         env::current_dir()?.join("results"),
     )?;
     if m.is_present("PUSH_IMAGE") {
-        let credentials = Credentials {
-            username: m.value_of("REGISTRY_USERNAME").unwrap(),
-            password: m.value_of("REGISTRY_PASSWORD").unwrap(),
-        };
-        docker_image.push(ui, &credentials)?;
+        match registry_type {
+            "amazon" => {
+                // The username and password should be valid IAM credentials
+                let provider = StaticProvider::new_minimal(
+                    m.value_of("REGISTRY_USERNAME").unwrap().to_string(),
+                    m.value_of("REGISTRY_PASSWORD").unwrap().to_string(),
+                );
+                // TODO TED: Make the region configurable
+                let client =
+                    EcrClient::new(default_tls_client().unwrap(), provider, Region::UsWest2);
+                let auth_token_req = GetAuthorizationTokenRequest { registry_ids: None };
+                let token = match client.get_authorization_token(&auth_token_req) {
+                    Ok(resp) => {
+                        match resp.authorization_data {
+                            Some(auth_data) => auth_data[0].clone().authorization_token.unwrap(),
+                            None => return Err(Error::NoECRTokensReturned),
+                        }
+                    }
+                    Err(e) => return Err(Error::TokenFetchFailed(e)),
+                };
+
+                let creds: Vec<String> = match base64::decode(&token) {
+                    Ok(decoded_token) => {
+                        match String::from_utf8(decoded_token) {
+                            Ok(dts) => dts.split(':').map(String::from).collect(),
+                            Err(err) => return Err(Error::Base64DecodeError(err)),
```

Comment:
I think this error might be a bit clearer if it was named differently, maybe something like `InvalidToken`?